aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -447,6 +449,155 @@ void qavg_pool2d_nhwc_kernel(
   });
 }
 
+template <typename T>
+void do_bilinear_on_AVX2(
+    const typename T::underlying*& pos1,
+    typename T::underlying*& pos2,
+    int64_t& c,
+    int64_t input_height,
+    int64_t input_width,
+    int64_t output_height,
+    int64_t output_width,
+    int64_t channels,
+    int32_t output_zero_point,
+    float multiplier,
+    const float h0lambda,
+    const float h1lambda,
+    const float w0lambda,
+    const float w1lambda,
+    const int64_t h1p,
+    const int64_t w1p) {
+#if defined(__AVX2__) && !defined(_MSC_VER)
+  constexpr auto vec_width = Vec256<T>::size() / 4;
+  if (vec_width == 8) {
+    for (; c + vec_width <= channels; c += vec_width) {
+      Vec256<float> pos1_fp_v[4];
+      Vec256<int32_t> pos1_int_v[4];
+      pos1_int_v[0] = vec256::convert_to_int32<typename T::underlying>(pos1);
+      pos1_int_v[1] = vec256::convert_to_int32<typename T::underlying>(
+          pos1 + w1p * channels);
+      pos1_int_v[2] = vec256::convert_to_int32<typename T::underlying>(
+          pos1 + h1p * input_width * channels);
+      pos1_int_v[3] = vec256::convert_to_int32<typename T::underlying>(
+          pos1 + (h1p * input_width + w1p) * channels);
+      for (int i = 0; i < 4; i++) {
+        int32_t pos1_int[vec_width];
+        float pos1_fp[vec_width];
+        pos1_int_v[i].store(pos1_int);
+        vec256::convert(pos1_int, pos1_fp, vec_width);
+        pos1_fp_v[i] = Vec256<float>::loadu(pos1_fp, 8);
+      }
+      Vec256<float> h0lambda_v(h0lambda);
+      Vec256<float> h1lambda_v(h1lambda);
+      Vec256<float> w0lambda_v(w0lambda);
+      Vec256<float> w1lambda_v(w1lambda);
+      Vec256<float> result =
+          h0lambda_v * (w0lambda_v * pos1_fp_v[0] + w1lambda_v * pos1_fp_v[1]) +
+          h1lambda_v * (w0lambda_v * pos1_fp_v[2] + w1lambda_v * pos1_fp_v[3]);
+      float result_fp[vec_width];
+      result.store(result_fp);
+      vec256::QuantizeAvx2<T>(
+          result_fp, pos2, vec_width, multiplier, output_zero_point);
+      pos1 += vec_width;
+      pos2 += vec_width;
+    }
+  }
+#endif
```

Comment:
Make an `#else` branch here with a `TORCH_INTERNAL_ASSERT(false` to catch bugs with the preprocessor conditionals in the future

Reply:
I think it is not supposed to fail in the else branch. We just ignore this code for non-AVX2 or windows build.